File: Java/LeetCode/Decode the Message.java
class Solution {
    
    public String decodeMessage(String key, String message) {
        StringBuilder decodedMessage = new StringBuilder();

        Map<Character, Character> keyMappings = new HashMap<>();
        char alphabetLetter = 'a';
        for (int i = 0; i < key.length(); ++i) {
            char c = key.charAt(i);
            if (!keyMappings.containsKey(c) && c != ' ') {
                keyMappings.put(c, alphabetLetter++);
            }
        }

        for (int i = 0; i < message.length(); ++i) {
            char c = message.charAt(i);
            decodedMessage.append( (c == ' ') ? c : keyMappings.get(c) );
        }

        return decodedMessage.toString();
    }
    
}

File: Kotlin/LeetCode/Decode the Message.kt
// https://leetcode.com/problems/decode-the-message

class Solution {

    fun decodeMessage(key: String, message: String): String {
        var decodedMessage: String = ""

        val keyMappings = mutableMapOf<Char, Char>()
        var alphabetLetter = 'a'
        for (i in 0 until key.length) {
            if (!keyMappings.containsKey(key[i]) && key[i] != ' ') {
                keyMappings.put(key[i], alphabetLetter++)
            }
        }

        for (i in 0 until message.length) {
            decodedMessage += if (message[i] == ' ') message[i] else keyMappings[message[i]]
        }

        return decodedMessage
    }

}
